File: frontend/src/services/feeService.js
import apiClient from './apiClient';

export const feeService = {
  getAllInvoices: async (params = {}) => {
    const response = await apiClient.get('/api/fees', { params });
    return response.data;
  },
  getInvoiceById: async (id) => {
    const response = await apiClient.get(`/api/fees/${id}`);
    return response.data;
  },
  getStudentInvoices: async (studentId) => {
    const response = await apiClient.get(`/api/fees/student/${studentId}`);
    return response.data;
  },
  createInvoice: async (data) => {
    const response = await apiClient.post('/api/fees', data);
    return response.data;
  },
  recordPayment: async (id, paymentData) => {
    const response = await apiClient.post(`/api/fees/${id}/pay`, paymentData);
    return response.data;
  },
  sendReminder: async (id) => {
    const response = await apiClient.post(`/api/fees/${id}/remind`);
    return response.data;
  },
};

File: backend/src/main/java/com/yourorg/appname/service/FeeService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.request.FeeInvoiceRequest;
import com.yourorg.appname.dto.request.PaymentRequest;
import com.yourorg.appname.dto.response.FeeInvoiceResponse;
import java.util.List;

public interface FeeService {
    List<FeeInvoiceResponse> getAllInvoices(String status, String query);
    FeeInvoiceResponse getInvoiceById(Long id);
    List<FeeInvoiceResponse> getInvoicesByStudentId(Long studentId);
    FeeInvoiceResponse createInvoice(FeeInvoiceRequest request);
    FeeInvoiceResponse recordPayment(Long id, PaymentRequest request);
    FeeInvoiceResponse sendReminder(Long id);
}
